JuliaGPU/session2/Aniruddh Patil/q1.jl

```diff
@@ -0,0 +1,55 @@
+#Sorting Algorithm Benchmark (Random Pivot Quick Sort vs. In-built Sort)
+using BenchmarkTools
+function quicksort(A,i=1,j=length(A))
+    if j > i
+        pivot = A[rand(i:j)]
+        left, right = i, j
+        while left <= right
+            while A[left] < pivot
+                left += 1
+            end
+            while A[right] > pivot
+                right -= 1
+            end
+            if left <= right
+                A[left], A[right] = A[right], A[left]
+                left += 1
+                right -= 1
+            end
+        end
+        quicksort(A,i,right)
+        quicksort(A,left,j)
+    end
+    return A
+end
+A=rand(2^20)
+@benchmark quicksort(A)
+@benchmark sort(A) 
```

Comment:
quicksort is a mutating function, so it sorts the contents of A. Thus when sort(A) is called for a sorted array A. So benchmark sort(A) first and then quicksort(A).